packages/augur-ui/src/modules/markets/actions/submit-new-market.ts

```diff
@@ -12,49 +12,77 @@ import { createBigNumber } from "utils/create-big-number";
 import { updateModal } from "modules/modal/actions/update-modal";
 import { MY_POSITIONS } from "modules/routes/constants/views";
 import { sortOrders } from "modules/orders/helpers/liquidity";
-import { addMarketLiquidityOrders } from "modules/orders/actions/liquidity-management";
+import { addMarketLiquidityOrders, clearMarketLiquidityOrders } from "modules/orders/actions/liquidity-management";
 import { AppState } from "store";
-import { NodeStyleCallback, NewMarket } from "modules/types";
+import { NodeStyleCallback, NewMarket, CreateMarketData } from "modules/types";
 import { ThunkDispatch } from "redux-thunk";
 import { Action } from "redux";
 import { createMarket } from "modules/contracts/actions/contractCalls";
 import { checkAccountAllowance } from "modules/auth/actions/approve-account";
+import { generateTxParameterId } from 'utils/generate-tx-parameter-id';
+import { generateTxParameters } from 'modules/create-market/helpers/construct-market-params';
+import { createMarketRetry } from "modules/contracts/actions/contractCalls";
 
 export function submitNewMarket(
```

Comment:
I'm not a fan of callers having to know what parameters need to be null. Create pass through methods to make it more clear what is being done in the calling code.